src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -8974,6 +8974,20 @@ and TcLookupThen cenv overallTy env tpenv mObjExpr objExpr objExprTy longId dela
         // To get better warnings we special case some of the few known mutate-a-struct method names
         let mutates = (if methodName = "MoveNext" || methodName = "GetNextArg" then DefinitelyMutates else PossiblyMutates)
 
+        if g.langVersion.SupportsFeature(LanguageFeature.InitPropertiesSupport) then
+            let minfo = List.head minfos
+
+            // Check, wheter this method has `IsExternalInit`, emit an error diagnostic in this case.
+            let hasInitOnlyMod =
+                match minfo with
+                | ILMeth (_, ilMethInfo, _) ->
+                    match ilMethInfo.ILMethodRef.ReturnType with
+                    | ILType.Modified(_, cls, _) -> cls.FullName = "System.Runtime.CompilerServices.IsExternalInit"
+                    | _ -> false
+                | _ -> false
+            if hasInitOnlyMod then
+                errorR (Error (FSComp.SR.tcGetterForInitOnlyPropertyCannotBeCalled1 methodName, mItem))
```

Comment:
```suggestion
                errorR (Error (FSComp.SR.tcSetterForInitOnlyPropertyCannotBeCalled1 methodName, mItem))
```

Btw, how do we know we're dealing with a setter here? Is that given by use of this attribute?

Reply:
We don't, we just rely on the attribute. By the C#'s spec this attribute can be on the setter only.